### PR TITLE
feat: menu component

### PR DESCRIPTION
**Pull request by hlavkat**

### CODE REVIEW

File: src/lib/components/ui/menu/menu-item.svelte [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	let { dimensions, children, ...restProps }: Props = $props()
+
+	const store = getContext<MenuStore>('menu-store')
+</script>
+
+<a {...restProps} class={dimensions ?? store.size}>
+	{#if children}
+		{@render children()}
+	{/if}
+</a>
```
Comment (vojtechsimetka) [inline src/lib/components/ui/menu/menu-item.svelte:17]: This should be both `button` and `a` element. Ideally, here we should just use `Button` component we already have

File: src/routes/+page.svelte [hunk 2]
```diff
@@ -548,6 +550,105 @@
...
+				<MenuItem href="#switch">Switch</MenuItem>
+				<MenuItem href="#typography">Typography</MenuItem>
+			</MenuTitle>
+			<MenuTitle bold content="Disabled" disabled>
+				<MenuItem href="#button">Button</MenuItem>
+				<MenuItem href="#checkbox">Checkbox</MenuItem>
+				<MenuItem href="#radio">Radio button</MenuItem>
+				<MenuItem href="#switch">Switch</MenuItem>
+				<MenuItem href="#typography">Typography</MenuItem>
+			</MenuTitle>
```
Comment (vojtechsimetka) [inline src/routes/+page.svelte:576]: Please add disabled variant

**Review by vojtechsimetka — CHANGES_REQUESTED** [review]

**Review by vojtechsimetka — APPROVED** [review]
LGTM, ready for final review @agazso

File: src/routes/+page.svelte [hunk 3]
```diff
@@ -548,6 +550,106 @@
 		</div>
 	</div>
 </section>
+<section>
+	<div class="row">
```
Comment (agazso) [inline src/routes/+page.svelte:554]: Missing section id and title:

```suggestion
<section id="menu">
        <Typography variant="h1">Menu</Typography>
	<div class="row">
```

**Review by agazso — APPROVED** [review]
Nice!

**Pull request merged**